### PR TITLE
feat: support maza

### LINKED ISSUE #332: Add Maza support
Hi,

will it be possible to add support for Maza Ad Blocker?

Link: https://github.com/tanrax/maza-ad-blocking

Description:
Like Pi-hole but local and using your operating system. Simple, native and efficient local ad blocker. Bash script compatible with MacOS, Linux, BSD and Windows Subsystem for Linux (WSL).

Updating is done by simply typing: maza update

### PR DESCRIPTION
**Pull request by SteveLauC**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The code compiles (`cargo build`)
- [x] The code passes rustfmt (`cargo fmt`)
- [x] The code passes clippy (`cargo clippy`)
- [x] The code passes tests (`cargo test`)
- [x] *Optional:* I have tested the code myself
    - [ ] I also tested that Topgrade skips the step where needed

If you developed a feature or a bug fix for someone else and you do not have the
means to test it, please tag this person here.

Closes #332

### CODE REVIEW

File: src/steps/os/unix.rs [hunk 1]
```diff
@@ -512,6 +512,13 @@ pub fn run_rcm(ctx: &ExecutionContext) -> Result<()> {
     ctx.run_type().execute(rcup).arg("-v").status_checked()
 }
 
+pub fn run_maza(ctx: &ExecutionContext) -> Result<()> {
+    let maza = require("maza")?;
+
+    print_separator("maza");
+    ctx.run_type().execute(maza).arg("update").status_checked()
```
Comment (DottoDev) [inline src/steps/os/unix.rs:519]: Return with ? at the eol and then the functions should return a Ok(())

**Review by DottoDev — CHANGES_REQUESTED** [review]

File: src/steps/os/unix.rs (hunk 1 above)
Comment (SteveLauC) [reply src/steps/os/unix.rs:519]: You mean something like this?
```rust
    ctx.run_type().execute(maza).arg("update").status_checked()?;
    Ok(())
}
```
Any reason why we should write it in this way?

Comment (DottoDev) [reply src/steps/os/unix.rs:519]: Could be that I'm mistaken but else if `require()` fails the `Err` doesn't get returned.

Comment (SteveLauC) [reply src/steps/os/unix.rs:519]: I am a little bit confused, why would the error derived from `require()` won't be returned? It is followed by a question mark `?`:
```rust
let maza = require("maza")?;
```

**Pull request merged**